src/core/container.jl

```diff
@@ -43,16 +43,23 @@ function Trace(m::Model, spl::T, vi::AbstractVarInfo) where {T <: AbstractSample
     res = Trace{T}(m, spl, deepcopy(vi));
     # CTask(()->f());
     res.vi.num_produce = 0
-    res.task = CTask( () -> begin vi_new=m(vi, spl); produce(Val{:done}); vi_new; end )
-    if isa(res.task.storage, Nothing)
+    res.task = CTask(() -> m(vi, spl))
```

Comment:
Do we need to do the same change for L35?

Reply:
Could you also briefly explain why this is needed?

Reply:
In other places (e.g. SMC https://github.com/TuringLang/Turing.jl/blob/master/src/inference/AdvancedSMC.jl#L51), `Val{:done}` is used to check if the running of the particles is finished.

Reply:
Nice catch! 

This is a small clean up after recent updates to `Libtask.jl`. In short, there were some caveats of previous task copying code, which prevents copied tasks exit normally. This is now fixed by `CTask`.